src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -33,15 +45,72 @@ def post(self, request, organization):
+        data = deepcopy(request.data)
+        # TODO: Possibly before creating the rule, or sometime - ensure at least one trigger is defined. Also ensure that each trigger has at least one action.
+        # If two triggers, ensure threshold constraints (warning triggers & resolves before critical)
+
+        alert_rule_serializer = AlertRuleSerializer(
+            context={"organization": organization, "access": request.access}, data=data
         )
+        if alert_rule_serializer.is_valid():
+            alert_rule = alert_rule_serializer.save()
+        else:
+            return Response(
+                {
+                    "AlertRuleError": alert_rule_serializer.errors,
+                    "message": "Could not create AlertRule",
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        triggers = data.get("triggers", [])
+        for trigger in triggers:
+            # TODO: Implement trigger constraints for thresholds (warning triggers & resolves before critical)
+            trigger_serializer = AlertRuleTriggerSerializer(
+                context={
+                    "organization": organization,
+                    "alert_rule": alert_rule,
+                    "access": request.access,
+                },
+                data=trigger,
+            )
+            if trigger_serializer.is_valid():
+                trigger_obj = trigger_serializer.save()
+            else:
+                # Could not create alert rule trigger.
+                # TODO: ROLL BACK ALERT_RULE CREATION (and trigger if any yet). DELETE IT?
+                return Response(
+                    {
+                        "AlertRuleTriggerError": trigger_serializer.errors,
+                        "message": "Could not create AlertRuleTrigger",
+                    },
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            actions = trigger.get("actions", [])
+            for action in actions:
+                action_serializer = AlertRuleTriggerActionSerializer(
+                    context={
+                        "organization": organization,
+                        "alert_rule": alert_rule,
+                        "trigger": trigger_obj,
+                        "access": request.access,
+                    },
+                    data=action,
+                )
 
-        if serializer.is_valid():
-            alert_rule = serializer.save()
-            return Response(serialize(alert_rule, request.user), status=status.HTTP_201_CREATED)
+                if action_serializer.is_valid():
+                    try:
+                        action = action_serializer.save()
+                    except InvalidTriggerActionError as e:
+                        # Could not create a trigger action
+                        # TODO: ROLL BACK ALL CREATIONS (Trigger(s), action(s) if any, and the alert rule)
```

Comment:
You can just wrap all these creations in a single transaction and it'll handle the rollback for you on exception. We probably don't want to have transaction logic in an endpoint, so we could make a function in logic.py to handle this. Or possibly in the serializer.